src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -480,13 +476,58 @@ private static ExpressionSyntax GetObjectCreationExpressionSyntax(Type type)
             else
             {
                 // Create an unformatted object.
-                Expression<Func<object>> getUninitializedObject =
+#if DNXCORE50
+                var bindingFlags = SyntaxFactoryExtensions.GetBindingFlagsParenthesizedExpressionSyntax(
+                    SyntaxKind.BitwiseOrExpression,
+                    BindingFlags.Instance,
+                    BindingFlags.NonPublic,
+                    BindingFlags.Public);
+                var nullLiteralExpressionArgument = SF.Argument(SF.LiteralExpression(SyntaxKind.NullLiteralExpression));
+                var typeArg = SF.Argument(SF.TypeOfExpression(typeInfo.GetTypeSyntax()));
+                var bindingFlagsArg = SF.Argument(bindingFlags);
+                var binderArg = nullLiteralExpressionArgument;
+                ArgumentSyntax ctorArgumentsArg;
+                var cons = typeInfo.GetConstructors(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+                    .OrderBy(p => p.GetParameters().Count()).FirstOrDefault();
+                var consParameters = cons != null ? cons.GetParameters().Select(p => p.ParameterType).ToArray() : null;
+                if (consParameters != null && consParameters.Length > 0)
+                {
+                    var separatedSyntaxList = new SeparatedSyntaxList<ExpressionSyntax>();
+                    separatedSyntaxList = consParameters
+                        .Aggregate(separatedSyntaxList, (current, t) => current.Add(SF.DefaultExpression(t.GetTypeInfo().GetTypeSyntax())));
+                    ctorArgumentsArg = SF.Argument(separatedSyntaxList.GetArrayCreationWithInitializerSyntax(
+                        SF.PredefinedType(SF.Token(SyntaxKind.ObjectKeyword))));
+                }
+                else
+                {
+                    ctorArgumentsArg = nullLiteralExpressionArgument;
+                }
+
+                var cultureInfoArg = SF.Argument(
+                    SF.IdentifierName("System").Member("Globalization").Member("CultureInfo").Member("InvariantCulture"));
+                var createInstanceArguments = new []
+                {
+                    typeArg,
+                    bindingFlagsArg,
+                    binderArg,
+                    ctorArgumentsArg,
+                    cultureInfoArg
+                };
+
+                Expression<Func<object>> getUninitializedObject = () => Activator.CreateInstance(default(Type));
+                result = SF.CastExpression(
+                    type.GetTypeSyntax(),
+                    getUninitializedObject.Invoke()
+                        .AddArgumentListArguments(createInstanceArguments));
+#else
+                Expression<Func<object>> getUninitializedObject = 
```

Comment:
Great :) Looks good to me.

On 15 December 2015 at 11:34, Dmitry Vakulenko <notifications@github.com>
wrote:

> In src/OrleansCodeGenerator/SerializerGenerator.cs
> <https://github.com/dotnet/orleans/pull/1122#discussion_r47581713>:
>
> > +                var createInstanceArguments = new []
> > +                {
> > +                    typeArg,
> > +                    bindingFlagsArg,
> > +                    binderArg,
> > +                    ctorArgumentsArg,
> > +                    cultureInfoArg
> > +                };
> > +
> > +                Expression<Func<object>> getUninitializedObject = () => Activator.CreateInstance(default(Type));
> > +                result = SF.CastExpression(
> > +                    type.GetTypeSyntax(),
> > +                    getUninitializedObject.Invoke()
> > +                        .AddArgumentListArguments(createInstanceArguments));
> > +#else
> > +                Expression<Func<object>> getUninitializedObject =
>
> I investigated ways of creating uninitialized object in CoreCLR and
> Activator.CreateInstance is the most suitable one. This approach was
> taken in https://github.com/kevin-montrose/Jil, and partially in
> https://github.com/JamesNK/Newtonsoft.Json.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/orleans/pull/1122/files#r47581713>.
>